### PR TITLE
fix: toolbox search final trigram

### LINKED ISSUE #2645: Toolbox search: long queries drop the last trigram, causing false positives (e.g. “conso” matches “math constrain”)
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Component

toolbox-search plugin (plugins/toolbox-search), version 3.1.3 (current main)

### Description

When searching in the toolbox search category, queries longer than three characters can match blocks that don’t contain the full query. Example: typing “conso” returns the built-in math_constrain block even though its text is “math constrain”. The trigram generator in src/block_searcher.ts stops one character early:

```
for (let start = 0; start < normalizedInput.length - 3; start++)
```

so “conso” produces only ["con", "ons"] (missing nso). Because matches are the intersection of generated trigrams, dropping the final trigram broadens results and yields false positives. Expected: include the final trigram (loop should use <=), so “conso” would not match “math constrain”.

I’m happy to work on a fix; the minimal change is to adjust the trigram loop in src/block_searcher.ts to include the final trigram (<= normalizedInput.length - 3).

### Reproduction steps

1. Install @blockly/toolbox-search and include a toolbox that contains the default math blocks (so math_constrain is present).
2. Add the search category to the toolbox: { kind: 'search', name: 'Search', contents: [] }.
3. Open the search category and type conso.
4. Observe that math_constrain appears in the results.

**Expected result**
No matches for “conso” unless a block actually contains that substring.

**Actual result**
math_constrain (and any block with “con”+“ons” but not the full “conso”) appears.

### Stack trace

```shell

```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by goughjo02**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/RaspberryPiFoundation/blockly-samples/issues/2645

### Proposed Changes

- Include the final trigram when generating search tokens to prevent partial matches on long queries.
- Add a regression test covering the “conso” false positive case.
- Expand BlockSearcher tests for short/empty input, underscore normalization, dropdown alt text indexing, and disambiguation between similar blocks.

### Reason for Changes

- Prevent toolbox search from returning false positives for queries longer than three characters and improve overall search accuracy.

### Test Coverage

- `npm test -- --runInBand` (blockly-scripts test)

### Documentation

- No documentation updates needed.

### Additional Information

- Additional tests define temporary blocks/dropdowns to exercise edge cases; no user-facing behavior changes beyond correcting search results.

### CODE REVIEW

File: plugins/toolbox-search/test/tests.mocha.js [hunk 1]
```diff
@@ -57,6 +66,119 @@ suite('BlockSearcher', () => {
...
+
+    const searcher = new BlockSearcher();
+    const blockInfo = {
+      kind: 'block',
+      type: 'searcher_underscore_block',
+    };
+    searcher.indexBlocks([blockInfo]);
+
+    assert.sameMembers(
+      searcher.blockTypesMatching('custom block with underscore'),
```
Comment (gonfunko) [inline plugins/toolbox-search/test/tests.mocha.js]: If the test is verifying that underscores in the block type get treated as spaces, should this be 'searcher underscore block'? As written this would match the unmodified message, but not verify the normalization of the block type.

File: plugins/toolbox-search/test/tests.mocha.js [hunk 2]
```diff
@@ -57,6 +66,119 @@ suite('BlockSearcher', () => {
...
+          type: 'searcher_dropdown_alt',
+          message0: 'weather %1',
+          args0: [
+            {
+              type: 'field_dropdown',
+              name: 'WEATHER',
+              options: [
+                [
+                  {
+                    src: 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEA',
```
Comment (gonfunko) [inline plugins/toolbox-search/test/tests.mocha.js]: This doesn't appear to actually be a valid image; if it doesn't otherwise cause problems, can you set `src` to `''` here and below to avoid confusion as to what this is and whether the particular value is relevant to the test case?

**Comment by gonfunko** [comment]
Thank you for identifying and reporting this issue and sending a fix! I really appreciate the improved tests as well. I just had two minor questions, and if you could run `npm run format` and commit the results of that to make the linter happy I think we should be good to merge this.

File: plugins/toolbox-search/test/tests.mocha.js (hunk 1 above)
Comment (goughjo02) [reply plugins/toolbox-search/test/tests.mocha.js]: I have update the code to verify the block type `searcher_underscore_block ` gets normalised

File: plugins/toolbox-search/test/tests.mocha.js (hunk 2 above)
Comment (goughjo02) [reply plugins/toolbox-search/test/tests.mocha.js]: i have deleted the contents of src strings. they are empty strings now

**Comment by gonfunko** [comment]
Thank you for making these changes, I really appreciate the fix!
